File: regexlib/2021-5-15/python_re_test_file/regexlib_4571.py
# 4571
# https?://[^<>() ]+([ ](((?!https?://)[^<>() ])+)(?=[^<>() ]*[?!=%&-+/])[^<>() ]*)*
# EXPONENT
# nums:4
# EXPONENT AttackString:"http://!"+" !!!"*32+"! _1_EOA(i or ii)"

import re
from time import perf_counter

regex = """https?://[^<>() ]+([ ](((?!https?://)[^<>() ])+)(?=[^<>() ]*[?!=%&-+/])[^<>() ]*)*"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "http://!" + " !!!" * i * 1 + "! _1_EOA(i or ii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_4571.js
/* 4571
 * https?://[^<>() ]+([ ](((?!https?://)[^<>() ])+)(?=[^<>() ]*[?!=%&-+/])[^<>() ]*)*
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"http://!"+" !!!"*32+"! _1_EOA(i or ii)"
 */
var REGEX = new RegExp("https?://[^<>() ]+([ ](((?!https?://)[^<>() ])+)(?=[^<>() ]*[?!=%&-+/])[^<>() ]*)*")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'http://!' + ' !!!'.repeat(i*1) + '! _1_EOA(i or ii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}